azure/durable_functions/models/DurableOrchestrationContext.py

```diff
@@ -466,6 +562,61 @@ def new_guid(self) -> UUID:
         guid = uuid5(NAMESPACE_URL, guid_name)
         return guid
 
+    @property
+    def _actions(self) -> List[List[Action]]:
+        """Get the actions payload of this context, for replay in the extension.
+
+        Returns
+        -------
+        List[List[Action]]
+            The actions of this context
+        """
+        if self._replay_schema is ReplaySchema.V1:
+            return self._action_payload_v1
+        else:
+            return [self._action_payload_v2]
+
+    def _add_to_actions(self, action_repr: Union[List[Action], Action]):
+        """Add a Task's actions payload to the context's actions array.
+
+        Parameters
+        ----------
+        action_repr : Union[List[Action], Action]
+            The tasks to add
+        """
+        # Do not add further actions after `continue_as_new` has been
+        # called
+        if self.will_continue_as_new:
+            return
+
+        if self._replay_schema is ReplaySchema.V1 and isinstance(action_repr, list):
+            self._action_payload_v1.append(action_repr)
+        elif self._replay_schema is ReplaySchema.V2 and isinstance(action_repr, Action):
+            self._action_payload_v2.append(action_repr)
+        else:
+            raise Exception(f"DF-internal exception: ActionRepr of signature {type(action_repr)}"
+                            f"is not compatible on ReplaySchema {self._replay_schema.name}. ")
+
+    def _produce_anonymous_task(self, parent: CompoundTask) -> TaskBase:
+        """Create an anonymous task, i.e one that isn't explicitely scheduled by the user.
+
+        This is to manage retryable tasks, where each retryable task may schedule
+        intermediate "anonymous" tasks such as timers as well as activities
```

Comment:
It's not immediately clear why we need these anonymous task wrappers to handle the CompoundTask scenario?

Reply:
Just for clarity: the anonymous tasks are **only** used for `RetryAbleTask`s and **not** for any other CompoundTask subclasses.

We only need them to create proxy tasks that represent the timers and re-scheduled tasks that occur on retryable tasks such as `callActivityWithRetry`. Happy to dig deeper into this on a call, and to modify the comments to communicate the intricacies of this bit.